agent/src/main/java/com/code_intelligence/jazzer/runtime/TraceDataFlowNativeCallbacks.java

```diff
@@ -19,47 +19,90 @@
 
 @SuppressWarnings("unused")
 final public class TraceDataFlowNativeCallbacks {
+  // Making this static final ensures that the JIT will eliminate the dead branch of a construct
+  // such as:
+  // if (USE_FAKE_PCS) ... else ...
+  private static final boolean USE_FAKE_PCS = useFakePcs();
+
   /* trace-cmp */
-  // Calls: void __sanitizer_cov_trace_cmp4(uint32_t Arg1, uint32_t Arg2);
-  public static native void traceCmpInt(int arg1, int arg2, int pc);
+  public static void traceCmpInt(int arg1, int arg2, int pc) {
+    if (USE_FAKE_PCS) {
+      traceCmpIntWithPc(arg1, arg2, pc);
+    } else {
+      traceCmpInt(arg1, arg2);
```

Comment:
Are the hooks really useful without fake PCs? How about only calling the CMP hooks only when `USE_FAKE_PCS` is set?

Reply:
I think that for integer compares the hooks don't even let the PC influence the way they add entries to the TORC, so I would expect them to still be useful (even without value profiling).

Reply:
You are right. The index is calculated by xoring the two arguments.